### PR TITLE
✨♻️ refactor preprocessing

### PR DESCRIPTION
**Pull request by dyollb**
## What do these changes do?

The PR adds 
- SelectChannel transform
- an option to crop the foreground using an otsu mask (thresholding the first channel)
- various small fixes
- improve `predict` command, to write metrics in csv file, plot summary confusion matrix

## How to test

- run pytest

## Checklist

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Unit tests for the changes and run locally with the command `pytest tests`
- [x] Runs on minimum Python version

### CODE REVIEW

File: src/segmantic/seg/losses.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    - "Unified Focal Loss: Generalising Dice and Cross Entropy-based Losses to Handle Class Imbalanced Medical Image Segmentation",
+    Michael Yeung, Computerized Medical Imaging and Graphics
+    - https://github.com/mlyg/unified-focal-loss/issues/8
+    """
+
+    def __init__(
+        self,
+        to_onehot_y: bool = False,
+        num_classes: int = 2,
+        gamma: float = 0.75,
```
Comment (dyollb) [inline src/segmantic/seg/losses.py:24]: gamma=2.0

File: src/segmantic/seg/losses.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+import torch
+from monai.networks import one_hot
+from monai.utils import LossReduction
+from torch.nn.modules.loss import _Loss
+
+
+class AsymmetricUnifiedFocalLoss(_Loss):
```
Comment (dyollb) [inline src/segmantic/seg/losses.py:9]: this should be AsymmetricFocalLoss:

see https://github.com/mlyg/unified-focal-loss/blob/411c9c5ce43b2ef847f0903d1b841512ad8d2eee/loss_functions.py#L193

to compute AsymmetricUnifiedFocalLoss we also need the AsymmetricTverskyLoss

see https://github.com/mlyg/unified-focal-loss/blob/411c9c5ce43b2ef847f0903d1b841512ad8d2eee/loss_functions.py#L351
